gerrit-entrypoint.sh

```diff
@@ -184,11 +180,40 @@ if [ "$1" = "/gerrit-start.sh" ]; then
   echo "Upgrading gerrit..."
   su-exec ${GERRIT_USER} java ${JAVA_OPTIONS} ${JAVA_MEM_OPTIONS} -jar "${GERRIT_WAR}" init --batch -d "${GERRIT_SITE}" ${GERRIT_INIT_ARGS}
   if [ $? -eq 0 ]; then
-    echo "Reindexing..."
-    su-exec ${GERRIT_USER} java ${JAVA_OPTIONS} ${JAVA_MEM_OPTIONS} -jar "${GERRIT_WAR}" reindex --verbose -d "${GERRIT_SITE}"
-    echo "Reindexing accounts..."
-    su-exec ${GERRIT_USER} java ${JAVA_OPTIONS} ${JAVA_MEM_OPTIONS} -jar "${GERRIT_WAR}" reindex --verbose --index accounts -d "${GERRIT_SITE}"
-    echo "Upgrading is OK."
+    GERRIT_VERSIONFILE="${GERRIT_SITE}/gerrit_version.txt"
+
+    if [ -z "${IGNORE_VERSIONCHECK}" ]; then
+      # dont perform a version check and never do a full reindex
+      NEED_REINDEX=0
+    else
+      # check whether its a good idea to do a full upgrade
+      NEED_REINDEX=1
+      echo "checking version file ${GERRIT_VERSIONFILE}"
+      if [ -f "${GERRIT_VERSIONFILE}" ]; then
+        OLD_GERRIT_VER="V"`cat ${GERRIT_VERSIONFILE}`
+        GERRIT_VER="V${GERRIT_VERSION}"
+        echo " have old gerrit version ${OLD_GERRIT_VER}"
+        if [ "${OLD_GERRIT_VER}" == "${GERRIT_VER}" ]; then
+          echo " same gerrit version, no upgrade necessary ${OLD_GERRIT_VER} == ${GERRIT_VER}"
+          NEED_REINDEX=0
+        else
+          echo " gerrit version mismatch #${OLD_GERRIT_VER}# != #${GERRIT_VER}#"
+        fi 
+      else
+        echo " gerrit version file does not exist, upgrade necessary"
+      fi
+    fi
+
+    if [ ${NEED_REINDEX} -eq 1 ]; then
+      echo "Reindexing all..."
+      su-exec ${GERRIT_USER} java ${JAVA_OPTIONS} ${JAVA_MEM_OPTIONS} -jar "${GERRIT_WAR}" reindex --verbose -d "${GERRIT_SITE}"
+    else 
+      echo "Reindexing accounts..."
+      su-exec ${GERRIT_USER} java ${JAVA_OPTIONS} ${JAVA_MEM_OPTIONS} -jar "${GERRIT_WAR}" reindex --verbose --index accounts -d "${GERRIT_SITE}"
```

Comment:
The re-index accounts action was a work around to reduce the restart time. Since we have a new way to determine if re-index is necessary or not I think it's safe to remove this line.